### DISCUSSION TITLE
How to create full-width table where first column is fixed size

### DISCUSSION THREAD

**Discussion by ssbarnea** [Textualize/rich#2878] [Q&A]
Apparently rich fails to respect the fixed column with on Tables configured to take the entire space (expand=True). 

I tried lots of combinations and I was never able to make Rich avoid expanding the size of the first column above its limits. This is quite problematic for anyone trying to display multiple tables with same columns one after another, as they would not be able to make their column sizes match.

```python
from rich.table import Table
from rich.console import Console


commands_table = Table(
    highlight=False,
    show_header=False,
    expand=True,
)
commands_table.add_column(
    style="bold cyan",
    width=10,
    min_width=10,
    max_width=10,
)
commands_table.add_column(
    "Description", style="bold red", justify="left", no_wrap=False
)

commands_table.add_row("a", "some")
commands_table.add_row("b", "some more")


console = Console()
console.print(commands_table)

```

The output is a table where the 1st column is also expanded,...

![](https://sbarnea.com/ss/Screen-Shot-2023-03-13-14-22-19.62.png)

**Comment by ssbarnea** [comment]
I think that I was able to find a weird workaround, adding big ratio value to the second column, makes it take all the space.

**Reply by willmcgugan** [reply]
You don't need a big ratio. A ratio of `1` will suffice. 

If you don't allow at least one column to grow, Rich will stretch all columns evenly with `expand=True`

**Reply by ssbarnea** [reply]
I am wondering if there is something we could to to provide a better UX for those wanting to to keep a column fixed.

Columns do not have an expand parameter, so that is implied on each of them. Setting max_width/min_width/width to a single value would count as a fixed column IMHO.

I wonder if adding an `expand` argument would make or not sense, one that would translate to making the ratio 0 or 1.

There is also the option to add more documentation and an example that covers exactly this use case.

Anyway, for the moment I was able to make a typer PR to improve its rendering, https://github.com/tiangolo/typer/pull/567


And a big thanks for all the work on rich!
